backend/local_oidc.ts

```diff
@@ -0,0 +1,414 @@
+import { Env } from "./types";
+import * as jose from "jose";
+import { v5 as uuidv5 } from "uuid";
+
+export interface OpenIdConfiguration {
+  issuer: string;
+  authorization_endpoint: string;
+  jwks_uri: string;
+  token_endpoint: string;
+  userinfo_endpoint: string;
+  end_session_endpoint: string;
+  scopes_supported: string[];
+  response_types_supported: string[];
+  token_endpoint_auth_methods_supported: string[];
+}
+
+interface UserData {
+  firstName: string;
+  lastName: string;
+  email: string;
+}
+
+export async function generatePEM(): Promise<string> {
+  // Use jose library to generate a key that can be exported as JWK
+  const { privateKey } = await jose.generateKeyPair("RS256", {
+    extractable: true,
+  });
+
+  // Export as PKCS8 PEM
+  const pem = await jose.exportPKCS8(privateKey);
+  return pem;
+}
+
+export async function ensurePEM(env: Env): Promise<string> {
+  await env.DB.prepare(
```

Comment:
👍

Reply:
Yeah, it's not the _best_ but I needed somewhere to shove the data for re-use. It won't affect production since we'll never turn this on there.